blob/gcsblob/gcsblob.go

```diff
@@ -107,14 +111,24 @@ func (o *lazyCredsOpener) OpenBucketURL(ctx context.Context, u *url.URL) (*blob.
 				o.err = err
 				return
 			}
+
+			// Populate default values from any such file, if and where possible.
```

Comment:
Can you add a link to the documentation where you found that this JSON blob might contain these fields? Maybe https://cloud.google.com/iam/docs/creating-managing-service-account-keys#creating_service_account_keys ?

Reply:
Appreciate your feedback to this PR so we can arrive at the best possible solution. I'll push a change asap, but reply to your points now so we're on the same page before the push will hide everything.

----
I'll add the link. I'll extract the “guessing” to its own function and cover the second example from aforementioned site.

I've asked around and received files without a private key (they have a ‘google’ sub-structure in them), so there's instances where only the `GoogleAccessID` is available.